honeybee/cli/edit.py

```diff
@@ -0,0 +1,292 @@
+        ratio: A number between 0 and 1 (but not perfectly equal to 1)
+            for the desired ratio between wall area and face area.
+        aperture_height: A number for the target height of the output apertures.
+            Note that, if the ratio is too large for the height, the ratio will
+            take precedence and the actual aperture_height will be larger
+            than this value.
+        sill_height: A number for the target height above the bottom edge of
+            the rectangle to start the apertures. Note that, if the
+            ratio is too large for the height, the ratio will take precedence
+            and the sill_height will be smaller than this value.
+        horizontal_separation: A number for the target separation between
+            individual aperture centerlines.  If this number is larger than
+            the parent rectangle base, only one aperture will be produced.
+    """
+    try:
+        assert os.path.isfile(model_json), 'No JSON file found at {}.'.format(model_json)
+        # serialize the Model to Python
+        with open(model_json) as json_file:
+            data = json.load(json_file)
+        parsed_model = Model.from_dict(data)
+        # check the Model tolerance
+        assert parsed_model.tolerance != 0, \
+            'Model must have a non-zero tolerance to use windows-by-ratio-rect.'
+        tol = parsed_model.tolerance
+
+        # generate the windows for all walls of rooms
+        for room in model.rooms:
+            for face in room.faces:
+                if isinstance(face.boundary_condition, Outdoors) and \
+                        isinstance(face.type, Wall):
+                    face.apertures_by_ratio_rectangle(
+                        ratio, aperture_height, sill_height, horizontal_separation,
+                        vertical_separation, tol)
+
+        # write the new model out to the file or stdout
+        output_file.write(json.dumps(parsed_model.to_dict()))
+    except Exception as e:
+        _logger.exception('Model windows by ratio rect failed.\n{}'.format(e))
+        sys.exit(1)
+    else:
+        sys.exit(0)
+
+
+@edit.command('overhang')
+@click.argument('model-json', type=click.Path(
+    exists=True, file_okay=True, dir_okay=False, resolve_path=True))
+@click.argument('depth', type=float)
+@click.option('--angle', help='A number for the for an angle to rotate the overhang '
+              'in degrees. Positive numbers indicate a downward rotation while '
+              'negative numbers indicate an upward rotation.',
+              type=float, default=0, show_default=True)
+@click.option('--vertical-offset', help='An optional number for the vertical offset of '
+              'the overhang from the top of the window or face. Positive numbers move '
+              'up while negative mode down.', type=float, default=0, show_default=True)
+@click.option('--by-window/--by-wall', help='Flag to note whether the overhangs should '
```

Comment:
I would suggest `per-window` and `per-wall`. It will be more explicit.

Reply:
Sounds good.  Will do.

Reply:
Done